utils/gen_onnx_mlir.py

```diff
@@ -260,95 +260,6 @@
     #("Transpose", "ImportNodeTranspose")
 ])
 
-# Operations supporting shape inference (alphabetical order).
```

Comment:
The list for ops with shape inference has not been used for a long time. I just deleted the list in this PR. 
I can move the translation of this OP to decompose. But there are other transformation in canonicalization and constant propagation that may replace an operation. I feel it is better not to assume the order of passes.  It would be better to start ShapeInference function with just `return success()` than  failure, because shape may be known or the operation can be eliminated with transformations.  I think that we can check the unsupported operations in the verification pass before lowering to Krnl.  
I tried to update the document of `HowToAddAnOperation.md` and found it is a big change that may attract lots of discussion. I will open another PR just for that part.